src/canvas.class.ts

```diff
@@ -1,1530 +1,1670 @@
-       * @param {CanvasRenderingContext2D} ctx Context to render controls on
-       */
-      drawControls: function (ctx) {
-        var activeObject = this._activeObject;
-
-        if (activeObject) {
-          activeObject._renderControls(ctx);
-        }
-      },
+  /**
+   * Draws objects' controls (borders/controls)
+   * @param {CanvasRenderingContext2D} ctx Context to render controls on
+   */
+  drawControls(ctx: CanvasRenderingContext2D) {
+    const activeObject = this._activeObject;
 
-      /**
-       * @private
-       */
-      _toObject: function (instance, methodName, propertiesToInclude) {
-        //If the object is part of the current selection group, it should
-        //be transformed appropriately
-        //i.e. it should be serialised as it would appear if the selection group
-        //were to be destroyed.
-        var originalProperties = this._realizeGroupTransformOnObject(instance),
-          object = this.callSuper(
-            '_toObject',
-            instance,
-            methodName,
-            propertiesToInclude
-          );
-        //Undo the damage we did by changing all of its properties
-        originalProperties && instance.set(originalProperties);
-        return object;
-      },
+    if (activeObject) {
+      activeObject._renderControls(ctx);
+    }
+  }
 
-      /**
-       * Realises an object's group transformation on it
-       * @private
-       * @param {fabric.Object} [instance] the object to transform (gets mutated)
-       * @returns the original values of instance which were changed
-       */
-      _realizeGroupTransformOnObject: function (instance) {
-        if (
-          instance.group &&
-          instance.group.type === 'activeSelection' &&
-          this._activeObject === instance.group
-        ) {
-          var layoutProps = [
-            'angle',
-            'flipX',
-            'flipY',
-            'left',
-            'scaleX',
-            'scaleY',
-            'skewX',
-            'skewY',
-            'top',
-          ];
-          //Copy all the positionally relevant properties across now
-          var originalValues = {};
-          layoutProps.forEach(function (prop) {
-            originalValues[prop] = instance[prop];
-          });
-          fabric.util.addTransformToObject(
-            instance,
-            this._activeObject.calcOwnMatrix()
-          );
-          return originalValues;
-        } else {
-          return null;
-        }
-      },
+  /**
+   * @private
+   */
+  _toObject(
+    instance: FabricObject,
+    methodName: 'toObject' | 'toDatalessObject',
+    propertiesToInclude: string[]
+  ): Record<string, any> {
+    // If the object is part of the current selection group, it should
+    // be transformed appropriately
+    // i.e. it should be serialised as it would appear if the selection group
+    // were to be destroyed.
+    const originalProperties = this._realizeGroupTransformOnObject(instance),
```

Comment:
i would prefer understand how i can add the transform to SVG rather than removing the transfrom from the object.
Probably if i spend 20 minute on this i can improve it, i all boil down to add another matrix on the object.